controller/controller_server/utils.py

```diff
@@ -359,3 +422,32 @@ def get_current_timestamp():
 
 def hash_string(string):
     return base58.b58encode(sha256(string.encode()).digest()).decode()
+
+
+def validate_parameters_match_volume(parameters, volume):
+    space_efficiency = parameters.get(config.PARAMETERS_SPACE_EFFICIENCY)
+    volume_space_efficiency = volume.space_efficiency
+    if space_efficiency:
+        space_efficiency = space_efficiency.lower()
+        if space_efficiency != volume_space_efficiency:
+            raise ValidationException(
+                messages.space_efficiency_not_match_volume_message.format(space_efficiency, volume_space_efficiency))
+    else:
+        if volume_space_efficiency not in [None, 'none', 'thick']:
```

Comment:
isn't this already checked in `validate_supported_space_efficiency`?

Reply:
e.g. we could change the new `Volume` field to `space_efficiency_synonyms` of type `set`, and check if the input is in it

Reply:
btw do we have a server unit test for this interesting `else` scenario?

Reply:
done, yes